src/components/ImageEditor/index.js

```diff
@@ -0,0 +1,259 @@
+    this.onMouseUp = this.onMouseUp.bind(this);
+
+    this.saveChanges = this.saveChanges.bind(this);
+  }
+
+  componentWillMount() {
+    document.addEventListener('mousemove', this.onMouseMove, false);
+    document.addEventListener('onmousemove', this.onMouseMove, false);
+    document.addEventListener('mouseup', this.onMouseUp, false);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this.onMouseMove);
+    document.removeEventListener('mouseup', this.onMouseUp);
+  }
+
+  /**
+   * Set initial selection area according to the maximum posible, and the landscape orientation
+   */
+  onImageLoad() {
+    const { ratio } = this.props;
+    const { clientWidth: width, clientHeight: height } = this.image;
+
+    const initialRatio = ratio || 1;
+    const landscape = width > height * initialRatio;
+
+    const selectionWidth = landscape ? (height * initialRatio) : width;
+    const selectionHeight = landscape ? height : (width / initialRatio);
+
+    const xOffset = landscape ? ((width - selectionWidth) / 2) : 0;
+    const yOffset = landscape ? 0 : ((height - selectionHeight) / 2);
+
+    this.setState(state => ({
+      ...state,
+      landscape,
+      x0: xOffset,
+      y0: yOffset,
+      x1: width - xOffset,
+      y1: height - yOffset,
+    }));
+  }
+
+  /**
+   * Takes the selection points relative to the image and transform it into css absolute properties relative to the wrapping div
+   * @return {Object} Object containing top, right, bottom and left css properties in percentage (or empty)
+   */
+  getSelectedAreaBoundaries() {
+    if (!this.container) {
+      return {};
+    }
+    const { x0, y0, x1, y1 } = this.state;
+    const { clientWidth: width, clientHeight: height } = this.container;
+    const { offsetLeft, offsetTop } = this.image;
+
+    return {
+      top: `${100*(Math.min(y0, y1) + offsetTop) / height}%`,
+      right: `${100*(width - Math.max(x0, x1) - offsetLeft) / width}%`,
+      bottom: `${100*(height - Math.max(y0, y1) - offsetTop) / height}%`,
+      left: `${100*(Math.min(x0, x1) + offsetLeft) / width}%`,
+    };
+  }
+
+  startMove(event) {
+    event.preventDefault();
+    const { clientWidth: width, clientHeight: height } = this.image;
+
+    const { pointX: xM, pointY: yM } = getClickPoint(event, this.container, this.image);
+
+    this.setState({
+      isMoving: true,
+      xM: minPositive(xM, width),
+      yM: minPositive(yM, height),
+    });
+  }
+
+  handleMove(event) {
+    event.preventDefault();
+    const { x0, y0, x1, y1, xM, yM } = this.state;
+    const { clientWidth: width, clientHeight: height } = this.image;
+
+    const clickPoint = getClickPoint(event, this.container, this.image);
+
+    const points = move(clickPoint, { height, width }, { xM, yM }, { x0, y0, x1, y1 });
+
+    this.setState(points);
+  }
+
+  startResize(event, invertX = false, invertY = false, isResizingSide = false, resizeVertical = false) {
```

Comment:
Could you refactor this to accept an object? It will be more meaningful to see function call in form `startResize({ invertX: true, invertY: true, ... })` instead of `startResize(true, true, ...).